File: cpp/cpp_100-999/829.cc
// C++
// leetcode 829
// https://leetcode.cn/problems/consecutive-numbers-sum/

class Solution {
public:
    int consecutiveNumbersSum(int n) {
    	int res = 0, cur = 1, sum = 1;
    	while(sum <= n) {
    		if((n-sum) % cur == 0)
    			res++;
    		cur++;
    		sum += cur;
    	}
    	return res;
    }
};

File: Go/Go_100-999/829.go
// Go
// leetcode 829
// https://leetcode.cn/problems/consecutive-numbers-sum/

func consecutiveNumbersSum(n int) int {
	res, cur, sum := 0, 1, 1
	for sum <= n {
		if (n-sum)%cur == 0 {
			res++
		}
		cur++
		sum += cur
	}
	return res
}